### DISCUSSION TITLE
259

### DISCUSSION THREAD

**Discussion by jojoldu** [jojoldu/blog-comments#597] [Comments]
http://jojoldu.tistory.com/259

**Comment by jojoldu** [comment]
작성자: 수

안녕하세요. 
올려주신 글 너무너무 유용하게 잘 보고있습니다. 

그런데, ec2에서 tomcat 같은 was를 별도로 설치하는 설명이 없던데
어떻게 하신건가요?? 
.war만  tomcat에 올려서 서비스를 하려고 하는데, 혹시 방법을 
아시나요??

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

안녕하세요!
5번째 시간인 http://jojoldu.tistory.com/263 
를 보시면 좋은 참고가 될것 같은데요.
이 프로젝트는 "스프링부트"로 진행되기 때문에 별도의 톰캣이 필요없습니다.
스프링부트의 경우에 프로젝트 내부에 톰캣을 가지고 있어(내장톰캣) 서버에 별도의 톰캣이 필요없습니다.

만약 스프링부트가 아닌, 기존 스프링+톰캣 구조로 가신다면 아래 링크를 참고하시면 될것 같습니다.

http://sarc.io/index.php/aws/520-aws-ec2-linux-10-apache-tomcat


추가로 스프링부트의 내장톰캣은 실서비스에는 못쓴다고 오해하실수도 있으실텐데요.
스프링부트는 하루 2천만 PV, 초당 1500 리퀘스트에서도 전혀 무리없이 사용했었기 때문에 외장 톰캣과 비교해서
 성능상 이슈는 전혀 고려하지 않으셔도 됩니다.

**Comment by jojoldu** [comment]
작성자: 수

스프링부트 였군요~~~

네네 너무너무 감사합니다. 링크해주신 부분 참고해서 해볼께요.
엄청엄청 감사합니다.

^^

**Comment by jojoldu** [comment]
작성자: soulduse@naver.com

이 부분은 파라미터 그룹으로 해결이 안되고, 직접 설정값을 적용해야만 합니다. (참고: 링크)

참고 링크가 깨진것 같아요~

**Comment by jojoldu** [comment]
작성자: 창천향로 창천향로

오 감사감사! 수정했어!

**Comment by chh9975** [comment]
pem 파일을 cp하면 파일을 찾지 못했다고 나옵니다
어떻게 해야 되나요??

**Comment by jojoldu** [comment]
@chh9975 메일 주신 분 맞으시죠? 같은 계정이셔서 다같이 보실수 있게 여기 쓰레드에 답글을 남깁니다.
pem 파일이 없다는 에러는 말 그대로 지정된 위치에 pem파일이 없다는건데요.
인스턴스 생성시 만든 pem파일이 실제로 존재하는지를 먼저 보셔야할것 같습니다.
어느 디렉토리로 다운받으신건가요?
만약 못찾으시겠다면, 다시 인스턴스 생성하셔서 pem 파일을 원하시는 곳에 정확히 다운 받으셔야합니다.
(잃어버린 pem 파일을 복구하는 방법을 제가 몰라서 ㅠㅠ...)

```no such file or directory permission denied (publickey)``` 역시 pem 위치를 몰라서 발생하는 이슈입니다.
pem파일을 어디에 다운받으셨는지 확인하시는게 먼저일것 같습니다.

**Comment by chh9975** [comment]
@jojoldu 답변 감사드립니다. 키페어 만들면서 pem파일을 다른곳으로 이동시켰는데 옮기면 안되는건가요?? 처음 파일을 다운받으면 다운로드 폴더로 떨어집니다.

**Comment by jojoldu** [comment]
@chh9975
음 본문대로 진행하실려면 pem 파일은 ```~/.ssh/``` 안에 있어야만 합니다.
다운로드 디렉토리는 어디든 상관없지만, pem을 받아서 ```~/.ssh```로 옮기셔야 합니다.
(맥/리눅스 PC에서만 가능하고 윈도우에선 잘 모르겠습니다 ㅠ )

**Comment by chh9975** [comment]
@jojoldu 네 알겠습니다 감사합니다 ㅠㅠ...
윈도우에서 옮기는 방법을 찾아봐야 겠네요...

**Comment by jojoldu** [comment]
@chh9975 앗 윈도우셨군요 ㅠ
[링크](http://pyrasis.com/book/TheArtOfAmazonWebServices/Chapter07/02) 를 참고하시면 도움이 되실것 같습니다.

**Comment by gonghojin** [comment]
잘 배우고 갑니다~ 좋은 글 감사합니다!!

**Comment by joont92** [comment]
대단하십니다..ㅋㅋ

**Comment by wwwkang8** [comment]
제가 우분투로 원격서버를 만들었는데 sudo yum install mysql 명령어가 작동하지 않는데 어떤 명령어를 써야하나요?

**Comment by dalek3** [comment]
좋은 글 감사합니다.

**Comment by jojoldu** [comment]
@wwwkang8 안녕하세요 우분투는 명령어가 yum이 아니라 apt-get 입니다
구글에서 우분투로 검색 해보셔야 할것 같습니다^^;

**Comment by jojoldu** [comment]
@joont92 앗 감사합니다 ㅎㅎ;;

**Comment by jojoldu** [comment]
@dalek3 봐주셔서 감사합니다^^;

**Comment by Jonny-Cho** [comment]
와 소화제 먹은 기분이에요. 답답함이 확 내려갔습니다. 감사합니다 :)

**Comment by ugaemi** [comment]
안녕하세요! 포스팅해주신 글 보면서 잘 따라하고 있습니다.
정말 자세하고 친절한 설명 감사드립니다 ㅎㅎ

**Comment by minseokLim** [comment]
keyPair가 없이는 ssh 접속이 불가능한데 굳이 ssh에 ip 제한을 두는 이유는 뭘까요? 혹시 모를 상황에 이중으로 대비한다고 생각하면 되나요? 저는 모바일 핫스팟으로 작업하는 중이라 그런지 잘 안되네요ㅠㅠ

**Comment by leahshinjekim** [comment]
안녕하세요, 따라해보면서 많은 도움을 받고 있습니다.
자세하고 친절한 포스팅 정말 감사드립니다!

**Comment by moonsiri** [comment]
인스턴스 탭을 클릭하셔서 탄력적 IP가 할당되었는지 확인합니다.

**Comment by moonsiri** [comment]
인스턴스 탭을 클릭하셔서 탄력적 IP가 할당되었는지 확인합니다.
밑에 사진 중간에 탄력적IP가 안가려졌네요ㅎㅎ

**Comment by IzacNocTurne** [comment]
Window 사용중 MySql Workbench로 AWS rds접속이안됩니다.
SSH Hostname: 엔드포인트 주소
SSH Username: RDS 마스터 사용자이름
MySQL Hostname: 127.0.0.1
MySQL Server Port: 3306
Username: root
SSH Key File에는 putty gen으로 만든 ppk를 pem으로 변환하여 첨부
일주일째 진전이없는데 해결 부탁드립니다. 감사합니다.
